client/views/account/accountProfile.coffee

```diff
@@ -38,7 +46,9 @@ Template.accountProfile.onCreated ->
 
 	@save = ->
 		instance = @
-
+		if !Settings.findOne("Accounts_AllowUsernameChange").value
+			toastr.error t('Username_Change_Disabled')
+			return callback()
```

Comment:
Seems that you don't need this callback here, probably will generate an error if the code hits this point.